lib/Clam/CfgBuilder.cc

```diff
@@ -2486,6 +2558,24 @@ void CrabIntraBlockBuilder::visitStoreInst(StoreInst &I) {
         m_bb.store_to_ref(ptr->getVar(), m_lfac.mkRegionVar(rgn), typed_const);
       }
     }
+
+    auto it = m_ref_fat_map.find(I.getPointerOperand());
+    if (it == m_ref_fat_map.end()) {
+      CLAM_WARNING("Could not find store ptr from map "
+                   << *I.getPointerOperand());
+    } else {
+      // check dereference is in bounds
+      // ref on operand:
+      // assert(offset >= 0)
+      // assert(obj size > offset)
+      CRAB_LOG("cfg-fat", llvm::errs()
+                              << "store: check size and offset are in bounds: "
+                              << I << "\n");
+      var_t ptr_obj_sz = it->second.first;
+      var_t ptr_offset = it->second.second;
+      m_bb.assertion(ptr_offset >= number_t(0), getDebugLoc(&I));
+      m_bb.assertion(ptr_obj_sz > ptr_offset, getDebugLoc(&I));
```

Comment:
It think this is not enough. I think we need to check that `ptr_offset + sizeof(I.getPointerOperand() < ptr_obj_sz`, where `sizeof` can be extracted from `DataLayout`